src/modules/evaluateExpression/evaluateExpression.ts

```diff
@@ -79,17 +70,21 @@ export default async function evaluateExpression(
         if (Object.entries(params).length === 0)
           return 'No parameters received for objectProperties node'
         try {
-          return params[childrenResolved[0].object][childrenResolved[0].property]
+          const objectIndex = childrenResolved[0].objectIndex ? childrenResolved[0].objectIndex : 0
+          const inputObject = params && params.objects ? params.objects[objectIndex] : {}
+          const property = childrenResolved[0].property
+          return extractNode(inputObject, property)
         } catch {
           return "Can't resolve object"
         }
 
       case 'pgSQL':
-        if (!params.pgConnection) return 'No database connection provided'
+        if (!params || !params.pgConnection) return 'No database connection provided'
         return processPgSQL(childrenResolved, query.type, params.pgConnection)
 
       case 'graphQL':
-        if (!params.graphQLConnection) return 'No database connection provided'
+        if (!params || !params.graphQLConnection) return 'No database connection provided'
```

Comment:
```suggestion
        if (!params.graphQLConnection) return 'No database connection provided'
```

Reply:
Hmm, I put it like this because I was getting a type error, since `params` could have additional properties. But it seems okay now that we've restricted the `IParameters` type more effectively. Nice, thanks :)